Disruption: Add param disruption

antoine-de opened the pull request:
add a 2 ways of filtering some disruptions:
- 1 on application period (with `datetime` and `duration` api params)
- 1 on publication period (with `publication_datetime`) (this one is more for debug and should not be used much)

to do more tests we need to add the publication_datetime param in ptref too

linked to http://jira.canaltp.fr/browse/NAVITIAII-1369

antoine-de committed:
Disruption: change disruption query from string to int

antoine-de committed:
Disruption: lint

antoine-de committed:
Disruption: add filter on period

antoine-de committed:
Disruption: add one test

antoine-de committed:
Disruption: add test on publication date filter

antoine-de committed:
Disruption: correction after review

kinnou02 commented on source/jormungandr/jormungandr/interfaces/v1/Disruptions.py:
```diff
@@ -65,11 +73,19 @@ def __init__(self):
+        parser_get.add_argument("datetime", type=date_time_format, default=datetime.now(),
+                                description="The datetime from which you want the disruption "
+                                            "(filter on the disruption application periods)")
+        parser_get.add_argument("publication_datetime", type=date_time_format, default=datetime.now(),
```
I don't really like the name, something like `current_datetime` would be better.
And it would be nice to explain that this parameter is mostly use for debug.

Finally, `now` or `utcnow`?

antoine-de committed:
Disruption: rename publciation_datetime api param

remaned to _current_datetime, since taht param will have to be added
everywhere

antoine-de committed:
Disruptions: add a TODO
